src/audio/base_fw.c

```diff
@@ -234,24 +236,46 @@ static int basefw_mem_state_info(uint32_t *data_offset, char *data)
 	return 0;
 }
 
+static log_timestamp_t basefw_get_timestamp(void)
+{
+	return sof_cycle_get_64() + time_delta;
+}
+
 static uint32_t basefw_set_system_time(uint32_t param_id,
 				       bool first_block,
 				       bool last_block,
 				       uint32_t data_offset,
 				       const char *data)
 {
-	/* TODO: configurate time to logging subsystem */
+	uint64_t host_time, host_time_cycle;
+	uint64_t current_dsp_time;
+	uint64_t current_dsp_cycle;
+
 	if (!(first_block && last_block))
 		return IPC4_INVALID_REQUEST;
 
 	global_system_time_info.host_time.val_l = ((const struct ipc4_system_time *)data)->val_l;
 	global_system_time_info.host_time.val_u = ((const struct ipc4_system_time *)data)->val_u;
 
-	uint64_t current_dsp_time = sof_cycle_get_64();
+	current_dsp_cycle = sof_cycle_get_64();
+	current_dsp_time = k_cyc_to_us_ceil64(current_dsp_cycle);
```

Comment:
This is changing the unit of `global_system_time_info.dsp_time`, why?

Reply:
both host & dsp time should be in us, not hw cycle.

Reply:
> both host & dsp time should be in us, not hw cycle.

If you're fixing a bug in `global_system_time_info.dsp_time`, then it must be described at the very least in the commit message of a separate and earlier commit (and even better: also described in a GitHub issue). Not secretly hidden in the addition of a new feature.

Reply:
sure, thanks

Reply:
@marc-hb https://github.com/thesofproject/sof/commit/da5edbdf7d1f73bfaccca0bfc2e19fe46097b0d0 add get/set ops for system time, they will be called on IPC, and could not be deleted. And this PR did something further, do some calculation for time delta, and use it to make log timestamp more accurate.